### PR TITLE
normalize Encryptor cert handling

### PR DESCRIPTION
**Pull request by kellyredding**
This ensures the given `opts[:cert]` is formatted as expected in
the constructor. This removes the need for a type-check in the
`openssl_cert` method and DRYs up removing any
`/-+(BEGIN|END) CERTIFICATE-+/` processing. This means you can
pass in `opts[:cert]` values in e.g. this form

```
-----BEGIN CERTIFICATE-----
ABC123...
-----END CERTIFICATE-----
```

and not get an `OpenSSL::X509::CertificateError nested asn1 error`
exception when calling `openssl_cert`.

### CODE REVIEW

**Comment by kellyredding** [comment]
Note: there were a bunch of trailing whitespace in this file my editor auto-removed. Let me know if you want me to take that out of the PR.

This is my first PR here so let me know if something more is needed. Thanks for considering!

**Comment by kellyredding** [comment]
Hey, just a friendly ping on this. Any thoughts @jphenow ? Not sure who to contact about PRs on this repo. Thanks.

**Comment by jphenow** [comment]
Heya - yea I don't get to invest all that much time in this these days but this seems like a reasonable change. Can you toss in some tests to the existing encryptor tests in `spec/`? I think that'll help illustrate/prove this a bit.

Thanks for the addition!

File: lib/saml_idp/encryptor.rb [hunk 1]
```diff
@@ -9,47 +9,43 @@ class Encryptor
...
     end
 
     def openssl_cert
-      if cert.is_a?(String)
-        @_openssl_cert ||= OpenSSL::X509::Certificate.new(Base64.decode64(cert))
-      else
-        @_openssl_cert ||= cert
-      end 
-    end 
+      @_openssl_cert ||= OpenSSL::X509::Certificate.new(Base64.decode64(cert))
```
Comment (Zogoo) [inline lib/saml_idp/encryptor.rb:32]: What's reason for this change. I think it might break that some peoples handling certificate object by themselves.

**Comment by Zogoo** [comment]
I’m closing this pull request since this can be done on the application side, and the pull request is quite old.

**Pull request closed without merging**